Dev/about

huan5678 opened the pull request:
此次施作項目:
1. 更新 關於我們頁面的RWD
2. 新增 FAQ頁面 及 聯絡我們 彈窗 表單 並接api 可以正確送信件 (均含RWD)
3. 套件增加 "bootstrap-icon",  "vue-validate", "yup"

huan5678 committed:
GIT COMMIT 訊息：

feat: 使用新圖標和常見問題解答功能增強用戶介面

- 添加 `bootstrap-icons` 套件至專案依賴項目
- 引入新的SVG組件 `about-header-desktop.svg`
- 引入新的SVG組件 `about-header-mobile.svg`
- 在主CSS檔案中導入 `bootstrap-icons.css`
- 在JSON格式中加入一個包含標題和問題的新常見問題解答部分
- 在路由配置中新增常見問題解答頁面的新路由
- 通過社交媒體連結和響應式設計改進增強關於頁面
- 創建新的FAQView組件，其中包含問題和答案的手風琴式折疊
- 用 `bootstrap-icons` 替換在關於、隱私以及條款視圖組件中的 `Asterisk` 圖標
- 更新隱私和條款視圖組件，以實現一致的圖標使用

Signed-off-by: huan5678 <huan5678@gmail.com>

huan5678 committed:
GIT 提交信息：

功能：實作帶有驗證的常見問題對話表單

- 在 `package.json` 依賴性中新增 `valibot` 與 `vee-validate` 套件
- 在 `public/img_components` 新增聯絡標題用的 SVG 圖片
- 在 `FAQView.vue` 中引入帶有用戶數據模式和表單處理的對話表單功能
- 在 `FAQView.vue` 的聯絡我們按鈕啟用對話顯示
- 在 `FAQView.vue` 對話中實作綁定 `valibot` 模式驗證的表單元素
- 在 `FAQView.vue` 對話表單中新增隱私政策的 `RouterLink`

Signed-off-by: 黃士桓 <huan5678@gmail.com>

huan5678 committed:
[wip]

huan5678 committed:
GIT 提交訊息：

重構：重構 FAQ 表單驗證和 API 邏輯

- 新增 `.env` 至 `.gitignore` 以避免提交環境設定檔
- 從 `package.json` 移除依賴項目 `@valibot/i18n`、`@vee-validate/valibot` 與 `valibot`
- 在 `package.json` 中加入新的依賴項目 `yup`
- 移除 `src/main.js` 中的 `valibot` 引入及相關的 i18n 設定
- 在 `src/views/FAQView.vue` 中重構表單驗證邏輯，使用 `yup` 替代 `valibot`
- 替換 `src/views/FAQView.vue` 中的表單處理和提交邏輯，使用 `axios` 進行 API 呼叫和 hooks 來進行成功/錯誤處理
- 在 `src/views/FAQView.vue` 中引入環境變數，用於 API 網址和聯繫郵件
- 更新 `src/views/FAQView.vue` 中的圖片尺寸類別以保持一致性
- 用基於 `yup` 的驗證架構取代 `src/views/FAQView.vue` 中的詳細驗證架構
- 在 `src/views/FAQView.vue` 中將表單欄位綁定至表單狀態和 `vee-validate` 邏輯

請記得翻譯所有提供的 git 提交訊息。

Signed-off-by: huan5678 <huan5678@gmail.com>

echocarriet commented:
我這邊沒問題唷 :)

huan5678 committed:
GIT 提交信息：

功能: 重構聯絡表單並更新用戶介面

- 新增一個 `ContactFormModal` 組件，實現具有驗證和提交邏輯的聯繫表單
- 在 `AboutView` 中，增加一個訂閱新聞通訊的表單，具有表單驗證，並使用 `ContactFormModal`
- 在 `FAQView` 中，移除頁面內的聯繫表單，改為使用 `ContactFormModal`
- 更新 `AboutView` 中的按鈕，以打開 `ContactFormModal`
- 修改 `AboutView` 中的新聞通訊訂閱表單，包括條款和隱私政策連結的更變
- 移除 `FAQView` 中與之前頁面內聯繫表單相關的未使用導入和代碼

Signed-off-by: huan5678 <huan5678@gmail.com>

sp1daTomm commented:
剛剛merge carrie 的pr 之後，這個pr有出現衝突

huan5678 committed:
GIT提交訊息：

功能：透過UI更新和依賴項目強化專案

- 新增`deploy.sh`腳本用於建置並部署專案到GitHub Pages
- 升級`axios`依賴版本，從`1.6.5`升至`1.6.7`
- 新增`tailwindcss-animated`和`vue-axios`依賴項目
- 為登入和註冊新增兩張新的使用者介面圖片
- 提升`NavbarEl.vue`導航欄類別的z-index值
- 新增`DelModal.vue`用於後台管理的刪除確認
- 新增`NavbarBackend.vue`和`NavbarVote.vue`用於後台和投票區域的導航
- 新增`ShareModal.vue`用於後台分享投票URL
- 在`main.js`啟動文件中整合`VueAxios`、`axios`及`bootstrap-icons`
- 新增`CollapseMixin.js`和`ModalMixin.js` mixins以利於模態框和摺疊元件的使用
- 在`index.js`中新增註冊、登入和管理後台的路由
- 新增登入(`LoginView.vue`)、註冊(`SignupView.vue`)、賬戶管理(`AccountView.vue`)、儀表板主頁(`DashboardView.vue`)和投票管理(`VoteView.vue`)的新視圖
- 在`tailwind.config.js`中整合`tailwindcss-animated`插件

Signed-off-by: 黃士桓 <huan5678@gmail.com>

huan5678 commented:
OK 修正衝突了

echocarriet commented:
抱歉，犯蠢！！
我這裡再次沒問題囉！

eee317 commented:
這個問題不大，後面有時間再調整就好
有發現彈窗的關閉按鈕沒有固定在最上面
然後捲軸到最底下後，背景會一起跟著滾動唷
![image](https://github.com/sp1daTomm/SelectWave/assets/38075054/1faa8cfd-1b5b-47c7-85a1-6db3d83e219f)

eee317 commented:
建議這邊的標題加上 py-3 的高度，有紅線的是我加入後的結果
有發現一些標題是 span ，可能 py-3 會沒有作用，要再加上 block 高度才會跑出來
![image](https://github.com/sp1daTomm/SelectWave/assets/38075054/1ba886ed-89c4-4191-920b-621bfc475358)

huan5678 committed:
GIT 提交訊息：

功能增強：重構 Vue 組件以實現手風琴式用戶介面

- 新增 Vue 組件 `AccordionComponent.vue`，用於創建手風琴式 UI 元素。
- 重構 `ContactFormModal.vue` 以移除對 `flowbite-vue` 組件的依賴，並直接使用 div 元素實現模態框功能，增加點擊模態框外部以及按 Esc 鍵關閉模態框的功能。
- 更新 `FAQView.vue`，使用新創建的 `AccordionComponent.vue` 代替 `flowbite-vue` 的手風琴組件，並增加計算屬性以處理 FAQ 數據。

Signed-off-by: 黃士桓 <huan5678@gmail.com>

huan5678 committed:
GIT提交訊息：

雜務：「重構項目依賴關係設置」

- 移除 `flowbite-vue` 和 `lucide-vue-next` 依賴
- 新增 `gsap` 為一個新的依賴
- 在依賴關係中將 `vee-validate` 重組至不同的順序
- 從 `main.css` 中移除對 `bootstrap-icons` 的匯入

Signed-off-by: 黃士桓 <huan5678@gmail.com>

huan5678 committed:
GIT 提交資訊：

feat: 改進使用者介面一致性和模態行為

- 在 `TopGoEl` 元件中將 `ChevronUp` 圖示元件替換為 `bi-chevron-up` 圖示類別
- 在 `AboutView` 的 `<main>` 元素中新增動態類別以處理模態顯示邏輯，以應對模態顯示時的溢出情況

Signed-off-by: 黃士桓 <huan5678@gmail.com>
